ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,182 @@
+ */
+
+const getActionsByEngineType = (type) => {
+  switch (type) {
+    case 'kv':
+      return [
+        {
+          actionTitle: 'Find KV secrets',
+          actionType: 'find-kv',
+        },
+      ];
+    case 'database':
+      return [
+        {
+          actionTitle: 'Generate credentials for database',
+          actionType: 'generate-credentials-db',
+        },
+      ];
+    case 'pki':
+      return [
+        {
+          actionTitle: 'Issue certificate',
+          actionType: 'issue-certificate-pki',
+        },
+        {
+          actionTitle: 'View certificate',
+          actionType: 'view-certificate-pki',
+        },
+        {
+          actionTitle: 'View issuer',
+          actionType: 'view-issuer-pki',
+        },
+      ];
+  }
+};
+
+const getActionParamByAction = (type) => {
+  switch (type) {
+    case 'find-kv':
+      return {
+        title: 'Secret Path',
+        subText: 'Path of the secret you want to read, including the mount. E.g., secret/data/foo.',
+        elementType: 'input',
+        buttonText: 'Read secrets',
+        path: 'vault.cluster.secrets',
+      };
+    case 'generate-credentials-db':
+      return {
+        title: 'Role to use',
+        elementType: 'search-select',
+        buttonText: 'Generate credentials',
+        model: 'database/role',
+        path: 'vault.cluster.secrets.backend.credentials',
+      };
+    case 'issue-certificate-pki':
+      return {
+        title: 'Role to use',
+        elementType: 'search-select',
+        placeholder: 'Type to find a role...',
+        buttonText: 'Issue leaf certificate',
+        model: 'pki/role',
+        path: 'vault.cluster.secrets.backend.pki.roles.role.generate',
+      };
+    case 'view-certificate-pki':
+      return {
+        title: 'Certificate serial number',
+        placeholder: '33:a3:...',
+        elementType: 'search-select',
+        buttonText: 'View certificate',
+        model: 'pki/certificate/base',
+        path: 'vault.cluster.secrets.backend.pki.certificates.certificate.details',
+      };
+    case 'view-issuer-pki':
+      return {
+        title: 'Issuer',
+        placeholder: 'Type issuer name or ID',
+        elementType: 'search-select',
+        buttonText: 'View issuer',
+        model: 'pki/issuer',
+        nameKey: 'issuerName',
+        path: 'vault.cluster.secrets.backend.pki.issuers.issuer.details',
+      };
+  }
+};
+
+export default class DashboardQuickActionsCard extends Component {
+  @service router;
+
+  @tracked selectedEngine;
+  @tracked selectedActions = [];
+  @tracked selectedAction;
+  @tracked actionParamField;
+  @tracked selectedEngineName;
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+
+    if (!this.selectedEngine) {
+      this.setSelectedAction();
+    }
+  }
+
+  get filteredSecretEngines() {
+    return this.args.secretsEngines.filter(
+      (secretEngine) =>
+        (secretEngine.shouldIncludeInList && secretEngine.type === 'pki') ||
```

Comment:
Maybe I'm reading this wrong, but I feel like from the name that 'shouldIncludeInList` should handle this filter and we shouldn't need the additional checks on the type. Maybe explain what `shouldIncludeInList` comes from or what List they're included in?

Reply:
This name is super confusing! `shouldIncludeInList` are all the secrets engines that are not excluded backends and we want to also filter that list to only show `pki`, `db`, and `kv` engines. Is it true that `pki`, `db`, and `kv` are never going to be excluded backends so I can just filter by those engine types?